# HG changeset patch
# User Sergey Mironov <grrwlf@gmail.com>
# Date 1379504718 -14400
#      Wed Sep 18 15:45:18 2013 +0400
# Node ID 748b6b4d60517e9f8098624a741b36c129385fc4
# Parent  923960c312bb9cdd74f8a31015c561d930e6104a
Add -print-ccompiler command line option

diff --git a/src/main.mlton.sml b/src/main.mlton.sml
--- a/src/main.mlton.sml
+++ b/src/main.mlton.sml
@@ -56,6 +56,8 @@
 		               raise Code OS.Process.success)
         fun printNumericVersion () = (print (Config.versionNumber ^ "\n");
 			              raise Code OS.Process.success)
+        fun printCCompiler () = (print ((Settings.getCCompiler ()) ^ "\n");
+			              raise Code OS.Process.success)
 
         fun doArgs args =
             case args of
@@ -67,6 +69,8 @@
               | "-css" :: rest =>
                 (css := true;
                  doArgs rest)
+              | "-print-ccompiler" :: rest =>
+                  printCCompiler ()
               | "-ccompiler" :: ccomp :: rest =>
                 (Settings.setCCompiler ccomp;
                  doArgs rest)
